File: CartagoFrontV2-master/src/app/services/referenciel/role.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Role } from '../../models/role';

@Injectable({
  providedIn: 'root',
})
export class RoleService {
  constructor(private http: HttpClient) {}

  getAllRoles(): Observable<Role[]> {
    return this.http.get<Role[]>('http://localhost:8080/Roles');
  }

  addRole(role: Role): Observable<Role> {
    return this.http.post<Role>('http://localhost:8080/Roles/', role);
  }

  updateRole(role: Role): Observable<Role> {
    return this.http.put<Role>('http://localhost:8080/Roles/', role);
  }

  deleteRole(id: number): Observable<Role> {
    return this.http.delete<Role>('http://localhost:8080/Roles/' + id);
  }
}

File: MicroServices/MicroServices/referenciel-service/src/main/java/com/bfi/referentiel/service/RoleService.java
package com.bfi.referentiel.service;

import com.bfi.referentiel.common.dto.RoleDTO;

import java.util.List;

public interface RoleService {

    void addRole(RoleDTO role);

    void updateRole(RoleDTO role);

    void deleteRole(Long id);

    List<RoleDTO> getAllRoles();
}
